src/main/java/org/springframework/data/redis/cache/RedisCache.java

```diff
@@ -181,7 +181,8 @@ public Boolean doInRedis(RedisConnection connection) throws DataAccessException
 			return null;
 		}
 
-		return new RedisCacheElement(cacheKey, fromStoreValue(lookup(cacheKey)));
+		Object storeValue = lookup(cacheKey);
```

Comment:
`lookup(…)` can return `null` values that don't indicate absence but can represent caching of a `null` value. This is because `lookup(…)` applies already deserialization.

We rather should split the actual `GET` from deserialization and inspect the result.